### DISCUSSION TITLE
Issues setting up redis

### DISCUSSION THREAD

**Discussion by Kulboi** [adonisjs/core#2422] [Help]
Hello,

I installed redis on my adonisjs backend using the following command **adonis install @adonisjs/redis** and added it to my providers in **start/app.js** like so;
const providers = [
 '@adonisjs/redis/providers/RedisProvider'
]

But I get the error: **Cannot find module '@adonisjs/redis/providers/RedisProvider'**. Also, I noticed no **config/redis** file was automatically created after installing the library. 

Would appreciate help on how to fix this.
Thank you.

**Comment by yarapolana** [comment]
You need to invoke it

`node ace invoke @adonisjs/redis`

this creates the config file and updates other necessary files.

More information [here](https://preview.adonisjs.com/guides/database/redis)

**Reply by Kulboi** [reply]
Thanks for the reply @yarapolana but i get this error when I run the command to invoke it;
**error: `invoke` is not a registered command**

**Reply by ebuka-broadshift** [reply]
I'm also getting this error error: invoke is not a registered command for webpack bundling.

**Comment by rafaelgo1705** [comment]
The table is happening to me, not only with redis, but also with other installation packages of Adonis 4.1

**Reply by artem-a** [reply]
@rafaelgo1705 Could you show your package.json file?

**Reply by rafaelgo1705** [reply]
{
  "name": "adonis-fullstack-app",
  "version": "4.1.0",
  "adonis-version": "4.1.0",
  "description": "The fullstack application boilerplate for Adonisjs",
  "main": "index.js",
  "scripts": {
    "start": "node server.js",
    "test": "node ace test"
  },
  "keywords": [
    "adonisjs",
    "adonis-app"
  ],
  "author": "",
  "license": "UNLICENSED",
  "private": true,
  "dependencies": {
    "@adonisjs/ace": "^5.0.8",
    "@adonisjs/auth": "^3.0.7",
    "@adonisjs/bodyparser": "~2.0.9",
    "@adonisjs/cors": "^1.0.7",
    "@adonisjs/fold": "^4.0.9",
    "@adonisjs/framework": "^5.0.9",
    "@adonisjs/ignitor": "^2.0.8",
    "@adonisjs/lucid": "^6.1.3",
    "@adonisjs/redis": "^2.0.7",
    "@adonisjs/session": "^1.0.27",
    "@adonisjs/shield": "^1.0.8",
    "@adonisjs/validator": "^10.0.0",
    "adonis-scheduler": "^3.0.2",
    "axios": "^0.21.1",
    "pg": "^8.5.1"
  },
  "devDependencies": {},
  "autoload": {
    "App": "./app"
  }
}

**Reply by artem-a** [reply]
Hmm, I think you have a problem with your "@adonisjs/validator" package, because you use version 10 which is not for V4.1. You need to change it on version 5.1.0 or legacy. Then remove your node_modules directory and re-install all the packages.

**Reply by rafaelgo1705** [reply]
Ok, thanks!

**Comment by artem-a** [comment]
@Kulboi What version of adonis do you use? And show me your package.json file.

**Comment by e-e** [comment]
Also having this issue
